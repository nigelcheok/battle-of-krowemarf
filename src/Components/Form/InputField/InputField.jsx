import React from 'react';

export function InputField(props) {
  return (
    <div className="form-group">
      <input
        type="text"
        className="form-control border-square"
        placeholder="Search for User"
        value={props.queryString}
        onChange={(e) => props.onQueryStringChanged(e.target.value)}
      />
    </div>
  );
}