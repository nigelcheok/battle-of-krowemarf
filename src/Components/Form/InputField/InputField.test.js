import React from 'react';
import { shallow } from 'enzyme';
import { InputField } from './InputField';

const clickFn = jest.fn();

describe('InputField', () => {
  it('should render correctly with no props', () => {
    const component = shallow(<InputField/>);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with props', () => {
    const component = shallow(<InputField onQueryStringChanged={clickFn}/>);

    component.find('input')
      .simulate('change', {target: {value: 't'}})
      .simulate('change', {target: {value: 'te'}})
      .simulate('change', {target: {value: 'tes'}})
      .simulate('change', {target: {value: 'test'}});

    expect(clickFn).toHaveBeenCalledTimes(4);
    expect(component).toMatchSnapshot();
  });
});