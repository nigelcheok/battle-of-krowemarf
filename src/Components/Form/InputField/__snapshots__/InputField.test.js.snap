// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField should render correctly with no props 1`] = `
<input
  className="form-control border-square"
  onChange={[Function]}
  placeholder="Search for user"
  type="text"
/>
`;

exports[`InputField should render correctly with props 1`] = `
<input
  className="form-control border-square"
  onChange={[Function]}
  placeholder="Search for user"
  type="text"
/>
`;
